aiida_quantumespresso/workflows/ph/base.py

```diff
@@ -95,6 +80,8 @@ def prepare_process(self):
         If a `restart_calc` has been set in the context, its `remote_folder` will be used as the `parent_folder` input
         for the next calculation and the `restart_mode` is set to `restart`.
         """
+        self.set_max_seconds(self.ctx.inputs.metadata.options['max_wallclock_seconds'])
```

Comment:
Won't this fail now in case the user hasn't specified the `max_wallclock_seconds`?

Reply:
Just tested this for the `PwBaseWorkChain`:
```Python
$ verdi process report 792
2021-05-06 07:31:54 [217 | REPORT]: [792|PwBaseWorkChain|on_except]: Traceback (most recent call last):
  File "/home/mbercx/.virtualenvs/aiida-dev/lib/python3.8/site-packages/plumpy/process_states.py", line 230, in execute
    result = self.run_fn(*self.args, **self.kwargs)
  File "/home/mbercx/envs/aiida-dev/code/aiida-core/aiida/engine/processes/workchains/workchain.py", line 214, in _do_step
    finished, stepper_result = self._stepper.step()
  File "/home/mbercx/.virtualenvs/aiida-dev/lib/python3.8/site-packages/plumpy/workchains.py", line 299, in step
    finished, result = self._child_stepper.step()
  File "/home/mbercx/.virtualenvs/aiida-dev/lib/python3.8/site-packages/plumpy/workchains.py", line 532, in step
    finished, result = self._child_stepper.step()
  File "/home/mbercx/.virtualenvs/aiida-dev/lib/python3.8/site-packages/plumpy/workchains.py", line 299, in step
    finished, result = self._child_stepper.step()
  File "/home/mbercx/.virtualenvs/aiida-dev/lib/python3.8/site-packages/plumpy/workchains.py", line 250, in step
    return True, self._fn(self._workchain)
  File "/home/mbercx/envs/aiida-dev/code/aiida-quantumespresso/aiida_quantumespresso/workflows/pw/base.py", line 392, in prepare_process
    self.set_max_seconds(self.ctx.inputs.metadata.options['max_wallclock_seconds'])
KeyError: 'max_wallclock_seconds'
```

Reply:
Good point. I could have sworn that the `max_wallclock_seconds` option was required but the `CalcJob` spec says otherwise. Maybe all scheduler implementations require it though?

Reply:
Just checked and it is not required, so this should indeed be fixed.